scripts/confirm_sns_subscription.py

```diff
@@ -0,0 +1,91 @@
+"""
+This script prompts the user for an SNS subscription confirmation link and then
+confirms the subscription with AuthenticateOnUnsubscribe enabled. Without
+AuthenticateOnUnsubscribe, anyone with an unsubscribe link would be able to
+delete the subscription. This would be especially undesired in the case of a
+subscription that sends notifications to a group email address.
+"""
+import logging
+
+from furl import (
+    furl,
+)
+
+from azul import (
+    config,
+)
+from azul.deployment import (
+    aws,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+
+log = logging.getLogger(__name__)
+
+sns_client = aws.sns
+
+
+def main() -> None:
+    if config.disable_monitoring:
+        print('SNS subscription confirmation skipped, monitoring is disabled.')
+    elif subscription_pending_confirmation():
+        confirm_subscription()
+    else:
+        print('SNS subscription confirmation skipped, no pending subscriptions found.')
+
+
+def subscription_pending_confirmation() -> bool:
+    """
+    Return True if there is at least one pending subscription
+    """
+    paginator = sns_client.get_paginator('list_subscriptions')
+    for page in paginator.paginate():
+        for subscription in page['Subscriptions']:
+            if subscription['SubscriptionArn'] == 'PendingConfirmation':
+                return True
+    return False
+
+
+def confirm_subscription() -> None:
+    """
+    Prompt user for a confirmation URL and process the confirmation.
+    """
+    # Expected URL format:
+    # https://sns.us-east-1.amazonaws.com/confirmation.html?TopicArn=...&Token=...&Endpoint=...
+    print()
+    print('This process will confirm a pending SNS subscription and configure\n'
+          'it so that an unsubscribe action requires AWS authentication.\n'
+          'Please open the "Subscription Confirmation" email, copy the\n'
+          '"Confirm subscription" link address, and paste the value here.\n'
+          'If you do not have a pending confirmation, this step can be skipped.')
+    url = input('URL: ').strip()
+    if url == '':
+        print('No URL given, skipping confirmation.')
+        return None
```

Comment:
No bail outs if possible. 

https://github.com/DataBiosphere/azul/blob/develop/CONTRIBUTING.rst#119control-flow

Furthermore, `return None` is the same as `return`.